### PR TITLE
Prepare to release v0.9.0

### PR DESCRIPTION
**Pull request by djs55**
- preliminary support for `Cstruct.t`
- add `topkg`
- modernize travis configuration

### CODE REVIEW

File: tar-unix.opam [hunk 1]
```diff
@@ -8,20 +8,18 @@ dev-repo:     "https://github.com/mirage/ocaml-tar.git"
...
-
-build-test: [
-  [ "jbuilder" "runtest" ]
+  ["jbuilder" "subst" "-p" name "--name" name] {pinned}
+  ["jbuilder" "build" "-p" name "-j" jobs]
 ]
 
 depends: [
-  "jbuilder"          {build}
+  "jbuilder"          {build & >="1.0+beta7"}
```
Comment (hannesm) [inline tar-unix.opam]: I likely have a stupid question, but do the other opam files not require a `jbuilder >= "1.0+beta7"`?

File: tar.opam [hunk 2]
```diff
@@ -8,11 +8,8 @@ dev-repo:     "https://github.com/mirage/ocaml-tar.git"
 doc:          "https://mirage.github.io/ocaml-tar/"
 
 build: [
-  [ "jbuilder" "build" "--only-packages=tar" ]
-]
-
-build-test: [
```
Comment (hannesm) [inline tar.opam]: this now means that the tests are never executed on travis!?

Comment (hannesm) [reply tar.opam]: ups, sorry -- tests seem to be executed in the `tar-mirage` opam -- do the tests depend on tar-mirage?

**Comment by hannesm** [comment]
apart from minor nitpicks, I think this is great!  thanks for working on this!

File: lib_test/jbuild [hunk 3]
```diff
@@ -1,6 +1,6 @@
 (executables
  ((names (parse_test))
-  (libraries (mirage-block-unix mirage-types-lwt oUnit lwt io-page.unix
+  (libraries (mirage-block-unix mirage-types-lwt oUnit lwt
```
Comment (hannesm) [inline lib_test/jbuild]: I guess travis wants to tell you that `io-page-unix` is still needed somewhere in here...

File: tar-unix.opam (hunk 1 above)
Comment (djs55) [reply tar-unix.opam]: I don't think that's a stupid question :) I resync'ed the `jbuild` files from `opam-repository` and only `tar-unix` seems to have that. I bet they should all have the `>= 1.0+beta7` really-- I'll adjust that.

File: tar.opam (hunk 2 above)
Comment (djs55) [reply tar.opam]: The `lib_test/parse_test.ml` uses `Tar_mirage` (not ideal, probably the tests should be split up)

**Comment by djs55** [comment]
One build is failing because `mirage-block-unix.2.8.3` doesn't build with `-safe-string`; fixed in https://github.com/mirage/mirage-block-unix/pull/83

I've added the `-safe-string` flag to the `jbuild` files here to be safe.

**Comment by djs55** [comment]
Since the only remaining problem is actually a failure of `mirage-block-unix` (fixed elsewhere) I think it's safe to do the release now.

**Pull request merged**